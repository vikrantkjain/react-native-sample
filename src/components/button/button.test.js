import React from 'react';
import {act} from 'react-test-renderer';
import Button from './index';
import {fireEvent, render, screen} from '@testing-library/react-native';
jest.mock('react-native-config', () => 'RNConfig');

const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('Button Testing with onPress event', () => {
  const onPressMock = jest.fn(() =>
    console.log('======== FUNCTION CALLED ========'),
  );
  const tree = render(
    <Button title="Title" onPress={onPressMock} disabled={false} />,
  );
  it('Render successfully', () => {
    expect(tree).toBeDefined();
  });
  it('Render correctly', () => {
    const treeToJSON = tree.toJSON();
    expect(treeToJSON).toMatchSnapshot();
  });
  act(() => {
    fireEvent.press(screen.getByTestId('BUTTON-TEST01'), eventData);
    expect(onPressMock);
  });
  it('Function not call when button is disabled', () => {
    const onPressDisabledMock = jest.fn(() => '=======DISABLED=======');
    render(
      <Button title="test" onPress={onPressDisabledMock} disabled={true} />,
    );
    fireEvent.press(screen.getByTestId('BUTTON-TEST01'));
    expect(onPressDisabledMock).not.toHaveBeenCalled();
  });
});
