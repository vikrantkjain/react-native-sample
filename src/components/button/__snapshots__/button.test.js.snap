// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing with onPress event Render correctly 1`] = `null`;
